Client/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -220,34 +222,24 @@ int CLuaResourceDefs::GetResourceConfig(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::GetResourceName(lua_State* luaVM)
+std::string CLuaResourceDefs::GetResourceName(lua_State* luaVM, std::optional<CResource*> resourceElement)
 {
-    // Verify arguments
-    CResource*       pResource = NULL;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
-
-    if (!argStream.HasErrors())
+    if (resourceElement && resourceElement.has_value())
     {
-        if (pResource)
-        {
-            // Grab its name and return it
-            const char* szName = pResource->GetName();
-            if (szName)
-            {
-                lua_pushstring(luaVM, szName);
-                return 1;
-            }
-        }
-        else
-            m_pScriptDebugging->LogBadPointer(luaVM, "resource", 1);
+        return (*resourceElement)->GetName();
     }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
 
-    // Failed
-    lua_pushboolean(luaVM, false);
-    return 1;
+    CLuaMain* localVM = m_pLuaManager->GetVirtualMachine(luaVM);
+
+    if (!localVM)
+        return false;
```

Comment:
The returning type is std::string. You cannot return bool here. Maybe you should use std::variant<std::string, bool> for the returning type. I know that's clumsy. The argument parser shall be improved before. So, okay, maybe it's better to choose std::invalid_argument, if you want to.

Reply:
`std::variant` is fine. Its long but its profitable